koku/api/report/test/ocp/test_ocp_query_handler.py

```diff
@@ -1531,3 +1535,254 @@ def test_format_tags(self):
+            if "values" in first_entry and len(first_entry["values"]) > 0:
+                first_value = first_entry["values"][0]
+                self.assertIn("model", first_value)
+
+    def test_gpu_group_by_vendor_returns_grouped_data(self):
+        """Test that GPU group_by vendor returns data grouped by vendor."""
+        url = reverse("reports-openshift-gpu")
+        params = {"group_by[vendor]": "*"}
+        url = f"{url}?{urlencode(params)}"
+        client = APIClient()
+        response = client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.data
+
+        # Verify group_by is in meta
+        self.assertIn("meta", data)
+        self.assertIn("group_by", data["meta"])
+        self.assertIn("vendor", data["meta"]["group_by"])
+
+        # Verify data structure has grouped values
+        self.assertIn("data", data)
+        if len(data["data"]) > 0:
+            first_entry = data["data"][0]
+            self.assertIn("date", first_entry)
+            if "values" in first_entry and len(first_entry["values"]) > 0:
+                first_value = first_entry["values"][0]
+                self.assertIn("vendor", first_value)
+
+    def test_gpu_order_by_without_matching_group_by_fails(self):
+        """Test that GPU order_by model without group_by model fails."""
+        url = reverse("reports-openshift-gpu")
+        params = {"order_by[model]": "asc"}
+        url = f"{url}?{urlencode(params)}"
+        client = APIClient()
+        response = client.get(url, **self.headers)
+
+        # Should return 400 Bad Request
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        data = response.data
+
+        # Verify error message mentions group_by requirement
+        self.assertIn("order_by", data)
+
+    def test_gpu_order_by_cost_with_group_by_model_works(self):
+        """Test that GPU order_by cost with group_by model works correctly."""
+        url = reverse("reports-openshift-gpu")
+        params = {"group_by[model]": "*", "order_by[cost]": "desc"}
+        url = f"{url}?{urlencode(params)}"
+        client = APIClient()
+        response = client.get(url, **self.headers)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        data = response.data
+
+        # Verify both group_by and order_by are in meta
+        self.assertIn("meta", data)
+        self.assertIn("group_by", data["meta"])
+        self.assertIn("order_by", data["meta"])
+        self.assertIn("model", data["meta"]["group_by"])
+
+    def test_gpu_filter_and_group_by_combination(self):
+        """Test that GPU filter and group_by work together correctly."""
+        with tenant_context(self.tenant):
+            # Get a model to filter by
+            models = list(
+                OCPGpuSummaryP.objects.values_list("gpu_model_name", flat=True).distinct().order_by("gpu_model_name")
+            )
+
+            if models:
```

Comment:
Do we need this `if` condition now that model bakery is loading test data for the gpu tables?
```suggestion

```